pkg/client/automation/client.go

```diff
@@ -120,6 +120,37 @@ func WithCustomUserAgentString(userAgent string) func(client *Client) {
 	}
 }
 
+// Get returns a single automation object matching the given resource type and id or an error
+// if a resource with the given ID and resourceType wasn't found
+func (a Client) Get(ctx context.Context, id string, resourceType ResourceType) (result *Response, err error) {
+	a.limiter.ExecuteBlocking(func() {
+		result, err = a.get(ctx, id, resourceType)
+	})
+	return
+}
+
+func (a Client) get(ctx context.Context, id string, resourceType ResourceType) (*Response, error) {
+	requestURL := a.url + a.resources[resourceType].Path + "/" + id
```

Comment:
please use path.Join, manual concatenation and adding/not adding slashes is always prone to break eventually

```suggestion
	requestURL := path.Join(a.url, a.resources[resourceType].Path, id)
```

Reply:
As we just need the URL string, maybe `url.JoinPath` could be enough and also replace the `url.Parse` call. 
Although it seems to only parse the basepath

Reply:
your suggestion does not play well with the host part of the URL, especially with the "http://" part and produces an invalid request url. I will jsut leave it like that

Reply:
I think we need to find a proper solution here, aka passing a proper URL object around instead of a plain string.
I recently had an issue where the concatenation caused a double `//`, thus failing requests. Would not have happened with path.Join.
I agree, maybe let's not fix this issue here, but we should be aware that we should not pass plain strings around.

Reply:
Since you're parsing the URL on line #134, you _could_ fix this issue fairly simple :)